### PR TITLE
Rename some package versions in Go discovery sample gen

### PR DESCRIPTION
**Pull request by tcoffee-google**
Resolves related compile-check errors.

### CODE REVIEW

**Comment by tcoffee-google** [comment]
This will also need to be adapted to Go MVVM.

File: src/main/java/com/google/api/codegen/go/GoDiscoveryContext.java [hunk 1]
```diff
@@ -159,21 +162,21 @@ public boolean isPageStreaming(Method method) {
...
   public String getApiVersion() {
-    String rename = API_VERSION_RENAME.get(getApi().getName(), getApi().getVersion());
-    return rename == null ? getApi().getVersion() : rename;
+    String version = getApi().getVersion();
+    if (version.equals("alpha") || version.equals("beta")) {
+      return "v0." + version;
+    }
+    Matcher subVersion = SUB_VERSION.matcher(version);
+    if (subVersion.matches()) {
+      return subVersion.group(1) + "/" + subVersion.group(2);
```
Comment (vchudnov-g) [inline src/main/java/com/google/api/codegen/go/GoDiscoveryContext.java]: Where is this slashed output used?

**Comment by vchudnov-g** [comment]
Fore reference here (and when doing MVVM, @saicheems):

See https://github.com/google/google-api-go-client/issues/170

The alpha and beta versions as well as api names of the form "apiname_v999" are remapped here: https://github.com/google/google-api-go-client/blob/master/google-api-go-generator/gen.go#L320

File: src/main/java/com/google/api/codegen/go/GoDiscoveryContext.java (hunk 1 above)
Comment (tcoffee-google) [inline src/main/java/com/google/api/codegen/go/GoDiscoveryContext.java]: IIRC, this case only arises in admin APIs, and not in our baselines.

Comment (vchudnov-g) [inline src/main/java/com/google/api/codegen/go/GoDiscoveryContext.java]: And the extra path component works fine in the imports? OK....

**Pull request merged**